src/v/cluster/topic_table.cc

```diff
@@ -831,6 +836,35 @@ void incremental_update(
     }
 }
 
+void incremental_update(
+  std::optional<model::shadow_indexing_mode>& property,
+  property_update<bool> overrides,
+  model::shadow_indexing_mode m) {
+    switch (overrides.op) {
+    case incremental_update_operation::remove: {
+        if (!property.has_value()) {
+            break;
+        }
+        auto simode = property.value();
+        property = model::add_shadow_indexing_flag(
+          simode, model::negate_shadow_indexing_flag(m));
+        return;
+    }
```

Comment:
Yes, you are correct, [due to the definition of `parse_and_set_bool()`](https://github.com/redpanda-data/redpanda/blob/4c2c24c468048905a00d4f52bc546e97ad88bcbf/src/v/kafka/server/handlers/configs/config_utils.h#L618-L625).

May be better to just assert here? (at the very least, the unused code path should be removed to not mislead readers as to behavior of the property here)

Reply:
I'm reluctant to go for vassert just because in case we ever decide to issue a remove in the future it would break this implementation. I think it's fine to keep the current implementation and maybe add a comment explaining that we don't currently use it.

Reply:
Added comment.